src/client/mod.rs

```diff
@@ -290,464 +215,122 @@ impl Client {
         Ok(Client {
             flags: flags,
             cluster_id: cluster_id,
-            mmv_path: mmv_path,
-            wi: MMVWriterInfo::new()
+            mmv_path: mmv_path
         })
     }
+    
+    pub fn export(&self, metrics: &mut [&mut MMVWriter]) -> io::Result<()> {
```

Comment:
> The compiler won't allow us to export the same metric concurrently

exactly, and since we are not establishing a 1:1 relationship between a metric and a client that reports it, we should probably avoid mutability. Also, from an implementation standpoint, for a single export we probably don't want the list of metrics exported to be mutable, or the metrics themselves